lib/facter/resolvers/partitions.rb

```diff
@@ -123,10 +123,10 @@ def populate_from_lsblk(partition_name, blkid_and_lsblk)
           return {} unless available?('lsblk', blkid_and_lsblk)
 
           lsblk_version_raw = Facter::Core::Execution.execute('lsblk --version 2>&1', logger: log)
-          lsblk_version = lsblk_version_raw.match(/ \d\.\d+/)[0].to_f
+          lsblk_version = lsblk_version_raw.match(/\d\.\d+/)&.to_f
```

Comment:
I think maybe?

```suggestion
          lsblk_version = lsblk_version_raw.match(/\d\.\d+/)&.first&.to_f
```

Reply:
I gave this a shot, but it didn't work. It's weird that MatchData acts like an Array in some ways (uses `[]` syntax) but does not support methods like `first`.

I tried another approach, let me know what you think